server/athenian/api/models/state/models.py

```diff
@@ -227,8 +227,31 @@ class ReleaseSetting(create_time_mixin(updated_at=True), Base):
     """Release matching rules per repo."""
 
     __tablename__ = "release_settings"
+    __table_args__ = (
+        UniqueConstraint(
+            "repo_id",
+            "logical_repo_name",
+            "account_id",
+            name="uc_release_settings_repo_id_logical_name_account",
+        ),
+    )
 
     repository = Column(String(), primary_key=True)
+
+    repo_id = Column(BigInteger(), nullable=True)
+    """The identifier of the repository the release setting is about.
+
+    Points to the mdb `Repository.node_id` column.
+    """
+
+    logical_repo_name = Column(String(), nullable=True)
```

Comment:
```suggestion
    logical_repo_name = Column(String(), primary_key=True)
```
Note: this implies `nullable=False`

Reply:
idea here is:
- add nullable columns
- run backfill
- some release settings will be "orphan" and will stay with repo_id NULL, fix this (by removing it I think)
- after having no NULLs, remove nullable, remove repository (meanwhile unused) and make repo_id,logical_name the primary key

Reply:
DEV-4929, DEV-4930